### ISSUE TITLE
[BUG] VM crash on map load

### ISSUE THREAD

**Issue by kcat** [ZDoom/gzdoom#1770]
### GZDoom version

g4.9pre-327-gef887403c

### Which game are you running with GZDoom?

Doom 2

### What Operating System are you using?

Linux x86_64

### Relevant hardware info

AMD Ryzen 5 3600, Radeon RX 580

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

With a recent GZDoom update, the engine started crashing in the VM when loading a map. Running with `+vm_jit false` to make sure it's not a JIT problem also crashes. I first noticed using Metadoom when it would load the TITLEMAP, but I've also tried without any mods where I can get to the title screen, but trying to start a game and load into a map crashes. I tried starting with a fresh ini, and it still crashes.

Using git bisect, I tracked it down to this commit:
```
453688ccc6d09edb6e36460425e48abbc0653c39 is the first bad commit
commit 453688ccc6d09edb6e36460425e48abbc0653c39
Author: Christoph Oelckers <coelckers@users.noreply.github.com>
Date:   Fri Oct 21 18:28:28 2022 +0200

    - reworked CVARs to not use a linked list and to be initialized manually.
    
    This solves two problems:
    
    * The linked list is too slow, a map is better. A map cannot be used with statically allocated CVARs because order of initialization is undefined.
    * The current CVAR system is an unordered mishmash of static variables and dynamically allocated ones and the means of identification are unsafe. With this everything is allocated on the heap so it can all be handled the same by the cleanup code.
```
The commit prior to this runs fine, but I get a crash when loading a map starting with that commit. Valgrind reports nothing of relevance until the crash, where it tries to read from address 0x48. Compiled with GCC 12.2. Here's a crash log:
[gzdoom-crash.log](https://github.com/ZDoom/gzdoom/files/9846059/gzdoom-crash.log)

### Steps to reproduce the behaviour.

Explain how to reproduce
1. 
2.
3.

**Comment by coelckers** [comment]
We already discussed this on Discord, it seems to be a problem specific to more recent GCC versions but currently do not know what makes things go wrong here.

Clang on Linux is fine, so is GCC 7.5.

**Comment by coelckers** [comment]
fixed in 2b805e196d2a418321528f5a4eea375ff9c5f0b9
